Added a few more programs on DS

shreayan98c opened the pull request.

SidJain1412 reviewed (changes requested):
Queue/SumOfPairs.py isn't a Queue question, plus it's brute force. Is there no better way to solve the problem?

shreayan98c commented:
I have fixed the folder structure. There was no existing folder for Hashmaps so I put it into Misc for now. Also added an efficient O(n) approach for the problem.

SidJain1412 commented:
@all-contributors please add @shreayan98c for code
